docs/current_docs/quickstart/agent/part2.mdx

```diff
@@ -0,0 +1,253 @@
+
+<Tabs groupId="language" queryString="sdk">
+    <TabItem value="go" label="Go">
+        Add the following new function to the main dagger module in `.dagger/main.go`:
+
+        ```go file=./snippets/part2/agent/go/agent.go.snippet
+        ```
+    </TabItem>
+    <TabItem value="python" label="Python">
+        Add the following new function to the main dagger module in `.dagger/src/hello_dagger/main.py`:
+
+        ```python file=./snippets/part2/agent/python/agent.py.snippet
+        ```
+    </TabItem>
+    <TabItem value="typescript" label="Typescript">
+        Add the following new function to the main dagger module in `.dagger/src/index.ts`:
+
+        ```typescript file=./snippets/part2/agent/typescript/agent.ts.snippet
+        ```
+    </TabItem>
+    <TabItem value="php" label="PHP">
+    Add the following new function to the main dagger module in `.dagger/src/HelloDagger.php`:
+
+    ```php file=./snippets/part2/agent/php/agent.php.snippet
+    ```
+    </TabItem>
+    <TabItem value="java" label="Java">
+        Add the following new function to the main dagger module in `.dagger/src/main/java/io/dagger/modules/hellodagger/HelloDagger.java`:
+
+        ```java file=./snippets/part2/agent/java/agent.java.snippet
+        ```
+
+        Be sure to add imports for the following:
+
+        ```Java
+import io.dagger.client.Client;
+import io.dagger.client.Env;
+import io.dagger.client.File;
+import io.dagger.client.LLM;
+import io.dagger.client.Workspace;
+        ```
+    </TabItem>
+</Tabs>
+
+This code creates a Dagger Function called `develop` that takes an assignment and codebase as input and returns a `Directory` with the modified codebase containing the completed assignment.
+
+There are a few important points to note here:
+- The variable `environment` is the environment to define inputs and outputs for the agent. Each input and output provides a description which acts as a declarative form of prompting.
+- Other Dagger module dependencies are automatically detected and made available for use in this environment. In this example, Dagger finds the `workspace` sub-module installed and automatically generates bindings for it: `with-workspace-input` and `with-workspace-output`. These bindings can be used to provide (or retrieve) the workspace to the agent.
+- The `environment` is privileged, which means that the environment includes all the Dagger Functions in the existing Daggerized CI pipeline, plus the core Dagger API. This allows the agent to use the existing `test` function to run unit tests.
+- The LLM is supplied with a prompt file instead of an inline prompt. By separating the prompt from the Dagger Function code, it becomes easier to maintain and update the instructions given to the LLM.
+- The `Env` in this agent is privileged, which means the environment will also include the current module and core Dagger APIs. This allows the agent to use the existing `test` function.
```

Comment:
```suggestion
```

Reply:
Ah I missed that you included that in the other bullets 👍